pkg/pillar/utils/cipherutils.go

```diff
@@ -43,7 +54,11 @@ func GetCipherCredentials(agentName string, status types.CipherBlockStatus) (typ
 			cipherBlock.Key(), err)
 		return handleCipherBlockCredError(agentName, cipherBlock, decBlock, err)
 	}
-	if err := proto.Unmarshal(clearBytes, &decBlock); err != nil {
+
+	var zconfigDecBlock zconfig.EncryptionBlock
+	err = proto.Unmarshal(clearBytes, &zconfigDecBlock)
+	decBlock = getEncryptionBlock(&zconfigDecBlock)
```

Comment:
I think ```getEncryptionBlock``` should be called after checking the proto un-marshaling error.

Reply:
Either way - we need the decBlock - we seem to be passing it to the handleError function?
Or can I just pass an empty decblock in that case?

Reply:
I will suggest passing empty decBlock in case of an error.

Reply:
Done..